pkg/controller/apmserver/apmserver_controller.go

```diff
@@ -227,6 +227,11 @@ func (r *ReconcileApmServer) isCompatible(as *apmv1beta1.ApmServer) (bool, error
 
 func (r *ReconcileApmServer) doReconcile(request reconcile.Request, as *apmv1beta1.ApmServer) (reconcile.Result, error) {
 	state := NewState(request, as)
+
+	if !association.IsConfigured(as, r.recorder) {
```

Comment:
We have to think about standalone APM server deployments, they will never have an association.

Reply:
If there's no association `IsConfigured` returns `true`:

```
// IsConfigured checks if an association is set in the spec and if it has been configured by an association controller.
```

Reply:
My bad! I should actually read the code and the docs :-(

Reply:
Naming was bad, I tried to improve the name of the method.

Reply:
Naming is hard! `SetAndConfigured` is interesting but it seems a bit odd for me. 
I like that the functions that return a boolean start with an `Is`.
Could be `IsSetAndConfigured` or `IsConfiguredIfSet`?

Reply:
I think I like `IsConfiguredIfSet`